fix(import-flow): optmize import flow api requests

sahil143 opened the pull request:
## Fixes 
<!-- For e.g Fixes: https://issues.redhat.com/browse/HAC-XXX -->
https://issues.redhat.com/browse/HAC-4812

## Description
<!-- Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change. -->


## Type of change
<!-- Please delete options that are not relevant. -->

- [ ] Feature
- [ ] Bugfix
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## Screen shots / Gifs for design review 
**Get UX suggestion on the modal text**
<!-- If change affects UI in any way, tag relevant UX people and add screenshots/gifs  -->
![Kapture 2023-12-12 at 18 08 58](https://github.com/openshift/hac-dev/assets/9278015/f3389910-9bbd-416e-b3c0-15e249634b9e)


## How to test or reproduce?
<!-- Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration -->

<!-- - [ ] Test A -->
<!-- - [ ] Test B -->

<!-- **Test Configuration(s)**: -->


## Browser conformance: 
<!-- To mark tested browsers, use [x] -->
- [ ] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge


<!-- ## Checklist: -->

<!-- 
- [ ] Code follows the style guidelines
- [ ] Self-reviewed the code
- [ ] Added comments in hard-to-understand areas
- [ ] Made corresponding changes to the documentation
- [ ] Changes generate no new warnings
- [ ] Added tests that prove this fix is effective or that the feature works
- [ ] New and existing unit tests pass locally with new changes
- [ ] Any dependent changes have been merged and published in downstream modules 
-->

Misjohns commented:
The placement of the helper text should be closer to the progress bar. 
See PF styling here: https://www.patternfly.org/components/progress#helper-text

Also, the length of the track (light blue bar) should be fixed. It seems to change based on the percentage being displayed.

rottencandy commented:
We should also probably remove the 'x' button since it doesn't do anything

![0](https://github.com/openshift/hac-dev/assets/20013884/53a97f43-97c4-4da3-be6d-9e25e7cc370a)

wolfeallison commented:
For the progress bar title, although "Hold onto your hats!!" made me laugh, we should change it to "Importing components" to match PatternFly's guidance https://www.patternfly.org/components/progress/design-guidelines/#content-considerations

sahil143 commented:
@rottencandy @Misjohns Addressed review comments.
Progress bar length was changing because of change in percentage outside the progress bar.

> The placement of the helper text should be closer to the progress bar.
See PF styling here: https://www.patternfly.org/components/progress#helper-text

There is no extra styling applied to the progress bar or the helper text. This is the styling coming from the patternfly.

<img width="953" alt="image" src="https://github.com/openshift/hac-dev/assets/9278015/9513d11c-adcd-421f-9bff-24df30cffc97">

rottencandy reviewed (approved):
/lgtm

The dispatch method seems a bit convoluted, but can't think of any other way to do it.

rohitkrai03 commented on src/components/ImportForm/GitImportForm.tsx:
```diff
@@ -67,11 +76,36 @@ const GitImportForm: React.FunctionComponent<React.PropsWithChildren<GitImportFo
+      const detectedComponents = values.selectedComponents
+        ? values.components?.filter((value, index) => values.selectedComponents[index])
+        : values.components;
+      if (detectedComponents.length > 5) {
```
I think it would be better if we show the modal all the time and be consistent with the experience.

rohitkrai03 reviewed (changes requested):
Overall works well. Let's just remove the condition of 5 or more components to show the progress modal.

rohitkrai03 commented on src/components/ImportForm/GitImportForm.tsx:
```diff
@@ -176,56 +212,81 @@ const GitImportForm: React.FunctionComponent<React.PropsWithChildren<GitImportFo
+      {!!maxProgressBar && !!submitModalText && (
+        <Modal
+          variant={ModalVariant.small}
+          title="Importing components"
```
Also consider adding a dynamic title for the modal based on number of components being imported.

rohitkrai03 reviewed (approved):
/approve
